sphinx/ext/napoleon/docstring.py

```diff
@@ -773,6 +785,108 @@ def _strip_empty(self, lines: List[str]) -> List[str]:
         return lines
 
 
+def _recombine_set_tokens(tokens):
+    def takewhile_set(iterable):
+        yield "{"
+
+        open_braces = 1
+        while True:
+            try:
+                token = next(iterable)
+            except StopIteration:
+                if open_braces != 0:
+                    raise ValueError("invalid value set: {}".format("".join(tokens)))
```

Comment:
this should probably become a warning

Reply:
@tk0miya, how would I emit that warning?

Reply:
Please use `logger.warning()` to emit a warning. `__()` is also helpful to make a warning message translatable.